From a4cd47de9bc22a0769dbda5bebf554a474328af0 Mon Sep 17 00:00:00 2001
Message-Id: <a4cd47de9bc22a0769dbda5bebf554a474328af0.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Tvrtko Ursulin <tvrtko.ursulin@intel.com>
Date: Tue, 28 Jan 2014 16:24:25 +0000
Subject: [PATCH 055/312] MUST_REBASE [VPG] drm/i915: Allow unprivileged
 userptr surfaces

Internal users of userptr want to create unsynchronized surfaces
from unprivileged user accounts. This is not acceptable in the
upstream world so we have to carry this patch forever. Luckily
it is a simple patch so shouldn't be too much of a burden.

Issue: VIZ-3438
Change-Id: If51d2a315e32bdd5fede7119d9d665bebcb85025
Signed-off-by: Tvrtko Ursulin <tvrtko.ursulin@intel.com>
Reviewed-on: https://android.intel.com/163407
Reviewed-by: Landowski, Piotr <piotr.landowski@intel.com>
Reviewed-by: Griffiths, Richard A <richard.a.griffiths@intel.com>
Reviewed-by: Odeberg, Hans <hans.odeberg@intel.com>
Tested-by: Odeberg, Hans <hans.odeberg@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_userptr.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_userptr.c b/drivers/gpu/drm/i915/i915_gem_userptr.c
index 21ea928..d71db83 100644
--- a/drivers/gpu/drm/i915/i915_gem_userptr.c
+++ b/drivers/gpu/drm/i915/i915_gem_userptr.c
@@ -280,7 +280,7 @@ i915_gem_userptr_init__mmu_notifier(struct drm_i915_gem_object *obj,
 	int ret;
 
 	if (flags & I915_USERPTR_UNSYNCHRONIZED)
-		return capable(CAP_SYS_ADMIN) ? 0 : -EPERM;
+		return 0; /* VPG: Allow normal users. */
 
 	down_write(&obj->userptr.mm->mmap_sem);
 	ret = i915_mutex_lock_interruptible(obj->base.dev);
@@ -339,7 +339,7 @@ i915_gem_userptr_init__mmu_notifier(struct drm_i915_gem_object *obj,
 		return -ENODEV;
 
 	if (!capable(CAP_SYS_ADMIN))
-		return -EPERM;
+		return 0; /* VPG: Allow normal users. */
 
 	return 0;
 }
-- 
1.7.9.5

